mobile/contacts.go

```diff
@@ -68,52 +69,12 @@ func (m *Mobile) ContactThreads(id string) (string, error) {
 	return toJSON(infos)
 }
 
-// ContactQueryEvent represents events generated by a contact query
-type ContactQueryEvent struct {
-	Type ContactQueryEventType    `json:"type"`
-	Data *core.ContactQueryResult `json:"data"`
-}
-
-// ContactQueryEventType is used to indicate a type of contact query event
-type ContactQueryEventType string
-
-const (
-	// ContactQueryData a match was found
-	ContactQueryData ContactQueryEventType = "data"
-	// ContactQueryDone query is complete or was canceled
-	ContactQueryDone = "done"
-)
-
-// FindContacts calls core FindContacts
-func (m *Mobile) FindContacts(query *core.ContactQuery, cb StringCallback) (func(), error) {
-	resCh, errCh, cancel := m.node.FindContacts(query)
-	doneFn := func() {
-		cb.Call(toJSON(ContactQueryEvent{
-			Type: ContactQueryDone,
-		}))
-	}
-	cancelFn := func() {
-		cancel.Close()
-		doneFn()
+// SearchContacts calls core SearchContacts
+func (m *Mobile) SearchContacts(query *pb.ContactQuery, options *pb.QueryOptions, cb Callback) (func(), error) {
```

Comment:
@asutula I changed this to use the search proto object, `QueryEvent`:
```
message QueryEvent {
    Type type        = 1;
    QueryResult data = 2;

    enum Type {
        DATA = 0;
        DONE = 1;
    }
}
```

Reply:
`QueryResult` has an `Any` type... curious if the JS impl. will be able to parse those into their correct types. Most likely, yes.